CONTRIBUTING.md

```diff
@@ -50,15 +50,15 @@ In order to run tests within the IDE, ensure the `Python Integrated Tools` / `Te
 
 Once your changes and tests are ready to submit for review:
 
-1. Test your changes
+1. Format your changes
 
-    Run the test suite to make sure that nothing is broken: `python3 setup.py test`.
+    Make sure that lint is passing by running `make lint`. To format your code, run `make format`.
 
-2. Format your changes
+    Consider using editor integrations to do it automatically: you'll need to configure [black](https://black.readthedocs.io/en/stable/integrations/editors.html) and [isort](https://github.com/PyCQA/isort/wiki/isort-Plugins).
 
-    Also make sure that lint is passing by running `make lint`. To format your code, run `make format`.
+2. Test your changes
 
-    Consider using editor integrations to do it automatically: you'll need to configure [black](https://black.readthedocs.io/en/stable/integrations/editors.html) and [isort](https://github.com/PyCQA/isort/wiki/isort-Plugins).
+    Run the test suite to make sure that nothing is broken: `make test`. You can also run integration tests with `make it` (they're much slower, though). To run everything including lint, use `make check-all`.
```

Comment:
I suggest we change the order here and require that `make check-all` passes *first* -- since these are the guidelines before raising a PR -- and then explain that to iterate faster `make test` and `make it` can also be used. In other words something like:

> Ensure that all tests pass by running `make check-all`. This runs sequentially lint checks, unit tests and integration tests. These can be executed in isolation using `make lint`, `make test` and `make it` respectively, in case you need to iterate over a subset of tests.
> 
>  Note: Integration tests are much slower than unit tests.

Reply:
Thanks, done.